content/chainguard/chainguard-images/getting-started/devcontainers/index.md

```diff
@@ -0,0 +1,183 @@
+---
+title: "Getting Started with the Chainguard Images as Dev Containers"
+type: "article"
+linktitle: "Dev Containers"
+aliases: 
+- /chainguard/chainguard-images/getting-started/devcontainers
+description: "Tutorial on using Chainguard Images as Dev Containers"
+date: 2024-04-15T11:07:52+02:00
+lastmod: 2024-04-15T11:07:52+02:00
+tags: ["Chainguard Images", "Products"]
+draft: false
+images: []
+menu:
+  docs:
+    parent: "getting-started"
+weight: 610
+toc: true
+---
+# Dev Containers with Chainguard Images
+The images based on Wolfi and maintained by Chainguard provide distroless images that can be used as Dev Containers with a few changes.
+
+Chainguard offers minimal runtime images designed for running production workloads, and development images that contain a shell and some development tooling. But both development and production images are slimmed down and updated regularly to be CVE free.
+
+Whilst it is obvious why production images should be secure, you may be wondering "why do I worry if my dev env gets hacked, there is no data in it?". The answer is simple. Your code goes into the production environment. If you have been hacked during development, then perhaps the hackers code goes into production as well. This is a "supply chain" attack.
+
+Hopefully you would notice a chunk of rogue code being added to your source. What you might not notice is a change to a dependency, that looks like an update, but is actually malware using a typo-squat. For example updating `color:1.1` to `co1or:1.2`. Especially in a huge list of dependency updates that you maybe don't even check. As long as your app works!
+
+We'll go through the changes you need to use a Chainguard development image as a Dev Container in VS Code.
+
+If you would like to build and use a Dev Container you will need a compatible Integrated Development Environment (IDE) or other tool. Here is list of [supported tools](https://containers.dev/supporting).
+
+To keep it simple, this guide is written for [Visual Studio Code (VS Code)](https://code.visualstudio.com/)
+
+You will also need a Docker server to connect to. Usually a local install of Docker Desktop, but full instructions can be found [here](https://code.visualstudio.com/docs/devcontainers/containers)
+
+
+{{< details "What is distroless" >}}
+{{< blurb/distroless >}}
+{{< /details >}}
+
+{{< details "What is Wolfi" >}}
+{{< blurb/wolfi >}}
+{{< /details >}}
+
+{{< details "Chainguard Images" >}}
+{{< blurb/images >}}
+{{< /details >}}
+
+## Simple setup
+
+You can get see a simple example if you take the content of the `chainguard-go-devcontainer` directory in our demo repository and push it to the root of an empty repository.
+
+For example:
+``` sh
+git clone https://github.com/chainguard-dev/edu-images-demos.git
+cd edu-images-demos/chainguard-go-devcontainer
+git init .
```

Comment:
So, I think it's got 2 jobs. The first part is a "Do this and it works" intro to Dev Containers. (Using Go)

The second part is more trying to explain the changes you need to make for using our images in a Dev Container. (Using Python). Root/nonroot is an entire topic in itself!

The different languages are used to avoid the inevitable "Does this work for other languages" question if the whole thing used Go. I did consider doing a 3rd section with another language but it would be mostly repeating the previous stuff with replacing Python with "whatever" and a different command to install dependencies. Maybe a different set of plugins in VSCode. It doesn't really add much. (Maybe replace all "python" with "node" and "pip" with "npm". Otherwise it looks nearly the same)


Pushing the folder back to a new repo means it works with GitHub Codespaces and you get prompted to "Reopen in Devcontainer" when you open the repo in VS Code. With it in a subdirectory, I don't think it ever prompts you. But it did say it's written for VS Code, so could maybe just make sure the commentary warns people that for Codespaces they'd have to move the folder to the root of their repo.